src/coreclr/src/jit/liveness.cpp

```diff
@@ -2435,24 +2435,25 @@ void Compiler::fgInterBlockLocalVarLiveness()
 
     for (varNum = 0, varDsc = lvaTable; varNum < lvaCount; varNum++, varDsc++)
     {
-        /* Ignore the variable if it's not tracked */
+        // Ignore the variable if it's not tracked
 
         if (!varDsc->lvTracked)
         {
             continue;
         }
 
-        if (lvaIsFieldOfDependentlyPromotedStruct(varDsc))
-        {
-            continue;
-        }
+        // Fields of dependently promoted structs may be tracked. We shouldn't set lvMustInit on them since
+        // the whole parent struct will be initialized; however, lvLiveInOutOfHndlr should be set on them
+        // as appropriate.
```

Comment:
If the field is dependently promoted (i.e. the whole struct is forced to live on the stack), then do we still need to mark them as `lvLiveInOutOfHndlr` since they'll always be defined/used to/from the stack?

Reply:
With this change we may enregister a field of a dependently promoted struct if the struct is no longer used. See the change in lclvars.cpp: we don't set lvaSetVarDoNotEnregister on such fields.